src/goose/brokers/redis/scheduler.clj

```diff
@@ -22,25 +24,39 @@
     (or (get-in job [:retry-opts :prefixed-retry-queue]) (:prefixed-queue job))
     (:prefixed-queue job)))
 
+(defn- sleep-duration [redis-conn scheduler-polling-interval-sec]
+  (let [total-process-count (heartbeat/total-process-count redis-conn)]
+    ; Sleep for total-process-count * polling-interval + jitters
+    ; Regardless of number of processes,
+    ; On average, Goose checks for scheduled jobs
+    ; every polling interval configured to reduce load on Redis.
+    ; All worker processes must have same polling interval.
+    (* 1000 (+ (* scheduler-polling-interval-sec total-process-count)
+               (rand-int 3)))))
+
+(defn- find-and-move-scheduled-jobs
+  "Returns truthy if due jobs were found."
+  [redis-conn]
+  (when-let [due-scheduled-jobs (redis-cmds/scheduled-jobs-due-now redis-conn d/prefixed-schedule-queue)]
+    (redis-cmds/enqueue-due-jobs-to-front redis-conn
+                                          d/prefixed-schedule-queue
+                                          due-scheduled-jobs
+                                          execution-queue)
+    true))
+
 (defn run
   [{:keys [internal-thread-pool redis-conn
            scheduler-polling-interval-sec]}]
   (log/info "Polling scheduled jobs...")
   (u/while-pool
     internal-thread-pool
     (u/log-on-exceptions
-      (if-let [jobs (redis-cmds/scheduled-jobs-due-now redis-conn d/prefixed-schedule-queue)]
-        (redis-cmds/enqueue-due-jobs-to-front
-          redis-conn d/prefixed-schedule-queue
-          jobs execution-queue)
-        ; Instead of sleeping when due jobs are found,
-        ; Goose immediately polls to check if more jobs are due.
-        (let [total-process-count (heartbeat/total-process-count redis-conn)]
-          ; Sleep for total-process-count * polling-interval + jitters
-          ; Regardless of number of processes,
-          ; On average, Goose checks for scheduled jobs
-          ; every polling interval configured to reduce load on Redis.
-          ; All worker processes must have same polling interval.
-          (Thread/sleep (* 1000 (+ (* scheduler-polling-interval-sec total-process-count)
-                                   (rand-int 3))))))))
+      (let [scheduled-jobs-found? (find-and-move-scheduled-jobs redis-conn)
```

Comment:
`enqueue-due-scheduled-jobs` & `enqueue-due-cron-entries` seem crisper names to me.